build/azure-pipelines/darwin/sql-product-build-darwin-signing.yml

```diff
@@ -18,13 +18,13 @@ steps:
     displayName: 'Rename the file'
 
   - task: UseDotNet@2
-    displayName: 'Install .NET Core sdk for signing'
+    displayName: 'Install .NET SDK for signing'
     inputs:
       packageType: sdk
-      version: 2.1.x
+      version: 6.0.x
```

Comment:
The 6.0 SDK is already installed, right? I'd just remove this then to simplify the pipeline - it's probably better to have it started failing if they remove the SDK from the agents anyways (since that likely implies it's out of date)

Reply:
The docs say it should - 

https://mssqltools.visualstudio.com/CrossPlatBuildScripts/_build/results?buildId=216716&view=logs&j=8bd3f1bd-7c30-5e86-3634-de6194e44401&t=713be44f-977c-4b4b-bc71-a1683d562305

```
Runner Image
Image: macos-12
Version: 20231115.2
Included Software: https://github.com/actions/runner-images/blob/macOS-12/20231115.2/images/macos/macos-12-Readme.md
Image Release: https://github.com/actions/runner-images/releases/tag/macOS-12%2F20231115.2
```
https://github.com/actions/runner-images/blob/macOS-12/20231115.2/images/macos/macos-12-Readme.md

```
.NET Core SDK: 6.0.417, 7.0.102, 7.0.202, 7.0.306, 7.0.404, 8.0.100
```

I notice in that run it still has the UseDotnet task - which has this line in the description : 

`Use this task to change the version of .NET Core used in subsequent tasks.`

so I'm assuming that's why. Could you try removing this and re-running to see if we still get a failure?

Reply:
Is this version tied to ESRP task or you could also use .NET 8 instead because it has a longer lifespan? (Hoping for .net fallback sdk to kick in)

Reply:
@Charles-Gagnon lemme try that; I figured that multiple .NET installations can be available simultaneously as on Windows, but that implies otherwise.

@cheenamalhotra `EsrpSigningTask@3` currently only supports up to .NET 6 (not even 7), to the consternation of other teams according to a PM over there.